### PR TITLE
Маврин Андрей 2536

### PR DESCRIPTION
**Pull request by shambala22**
Features:
- Showing 60 last images from Yandex photo of the day.(portrait - 6 on page, landscape - 10 on page)
- Showing full-screen images
- Caching downloaded images
- Updating by swipe down, progress bar
- Buttons for saving image in gallery setting it as wallpaper

App downloads images in low resolution ~(200x300), because higher resolution's downloading takes a lot of time. But you can change image size by changing one constant in IntentService.

### CODE REVIEW

**Comment by thevery** [comment]
Actually you should download small images for thumbnails and large for fullscreen

**Comment by shambala22** [comment]
Really. I don't know, why i didn't think about it.
Now I did it. 
App downloads high-resolution pics after first click on it, and loads that pics from cache after second and consequent clicks. 
Also I added opportunity to open page with image in browser.

File: app/src/main/java/shambala/md/ifmo/photoofthday/ImageUpdater.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+
+/**
+ * Created by 107476 on 10.01.2015.
+ */
+public class ImageUpdater extends IntentService {
+    public static final String url = "http://api-fotki.yandex.ru/api/podhistory/?format=json";
```
Comment (svasilinets) [inline app/src/main/java/shambala/md/ifmo/photoofthday/ImageUpdater.java:30]: doesn't comply with java code style

File: app/src/main/java/shambala/md/ifmo/photoofthday/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+     * loaded fragment in memory. If this becomes too memory intensive, it
+     * may be best to switch to a
+     * {@link android.support.v4.app.FragmentStatePagerAdapter}.
+     */
+    SectionsPagerAdapter mSectionsPagerAdapter;
+    SectionsLandscapePagerAdapter mSectionsLandscapePagerAdapter;
+    /**
+     * The {@link ViewPager} that will host the section contents.
+     */
+    ViewPager mViewPager;
```
Comment (svasilinets) [inline app/src/main/java/shambala/md/ifmo/photoofthday/MainActivity.java:64]: visibility modifiers?

File: app/src/main/java/shambala/md/ifmo/photoofthday/OneImage.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        progressBar.setVisibility(View.GONE);
+        index = getIntent().getIntExtra("index", 0);
+        mReceiver = new ImagesReceiver(new Handler());
+        mReceiver.setReceiver(this);
+        try {
+            FileInputStream fis = this.openFileInput("" + index);
+            b = BitmapFactory.decodeStream(fis);
+            fis.close();
+        } catch (Exception e) {
+            Toast.makeText(this, "Can't load image", Toast.LENGTH_SHORT).show();
```
Comment (svasilinets) [inline app/src/main/java/shambala/md/ifmo/photoofthday/OneImage.java:68]: All ui strings must be in resources

File: app/src/main/java/shambala/md/ifmo/photoofthday/OneImage.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        updated = getIntent().getBooleanExtra("updated", false);
+        progressBar = (ProgressBar) findViewById(R.id.progressBar2);
+        progressBar.setMax(100);
+        progressBar.setVisibility(View.GONE);
+        index = getIntent().getIntExtra("index", 0);
+        mReceiver = new ImagesReceiver(new Handler());
+        mReceiver.setReceiver(this);
+        try {
+            FileInputStream fis = this.openFileInput("" + index);
+            b = BitmapFactory.decodeStream(fis);
```
Comment (svasilinets) [inline app/src/main/java/shambala/md/ifmo/photoofthday/OneImage.java:65]: IO operations with filesystem in UI thread are prohibited!

File: app/src/main/java/shambala/md/ifmo/photoofthday/ImageUpdater.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                newImages.add(i, image);
+                ContentValues values = new ContentValues();
+                values.put("ind", i);
+                values.put("link", hLink);
+                values.put("onsite", onSiteLink);
+                getContentResolver().insert(ImagesContentProvider.LINKS_CONTENT_URI, values);
+                saveImage(image, i);
+
+            }
+            MainActivity.downloaded = newImages;
```
Comment (svasilinets) [inline app/src/main/java/shambala/md/ifmo/photoofthday/ImageUpdater.java:99]: It is bad to have static fields in Activity, it is super bad to access them from a service

**Comment by svasilinets** [comment]
Almost ok
- 1 (inaccurate code: strings in code and etc)
- 2 (static field in the Activity accessed by your Service)
- 1 (IO operations in a main thread)
- 1 (Caching of large images works badly: if you restart your app without a connection, earlier loaded large images aren't shown)

Result: 20 - 5  = 15.

**Pull request closed without merging**